File: Readio/src/main/java/ninegle/Readio/subscription/mapper/SubscriptionMapper.java
package ninegle.Readio.subscription.mapper;

import org.springframework.stereotype.Component;

import ninegle.Readio.subscription.domain.Subscription;
import ninegle.Readio.subscription.dto.response.SubscriptionResponseDto;

@Component
public class SubscriptionMapper {

	public SubscriptionResponseDto toDto(Subscription subscription) {
		return SubscriptionResponseDto.builder()
			.userId(subscription.getUserId())
			.subDate(subscription.getSubDate())
			.expDate(subscription.getExpDate())
			.active(subscription.isActive())
			.canceled(subscription.isCanceled())
			.build();
	}
}

File: Readio/src/main/java/ninegle/Readio/subscription/mapper/SubscriptionMapper.kt
package ninegle.Readio.subscription.mapper

import ninegle.Readio.subscription.domain.Subscription
import ninegle.Readio.subscription.dto.response.SubscriptionResponseDto
import org.springframework.stereotype.Component

@Component
class SubscriptionMapper {
    fun toDto(subscription: Subscription): SubscriptionResponseDto =
        SubscriptionResponseDto(
            userId = subscription.userId,
            subDate = subscription.subDate,
            expDate = subscription.expDate,
            active = subscription.isActive(),
            canceled = subscription.canceled
        )
}